src/traces/scatterternary/hover.js

```diff
@@ -17,12 +17,29 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
     var scatterPointData = scatterHover(pointData, xval, yval, hovermode);
     if(!scatterPointData || scatterPointData[0].index === false) return;
 
+    var newPointData = scatterPointData[0];
+
     // if hovering on a fill, we don't show any point data so the label is
-    // unchanged from what scatter gives us.
-    if(scatterPointData[0].index === undefined) return scatterPointData;
+    // unchanged from what scatter gives us - except that it needs to
+    // be constrained to the trianglular plot area, not just the rectangular
+    // area defined by the synthetic x and y axes
+    // TODO: in some cases the vertical middle of the shape is not within
+    // the triangular viewport at all, so the label can become disconnected
+    // from the shape entirely. But calculating what portion of the shape
+    // is actually visible, as constrained by the diagonal axis lines, is not
+    // so easy and anyway we lost the information we would have needed to do
+    // this inside scatterHover.
```

Comment:
This TODO is fairly uncommon, but when it happens you can see things like this:
![screen shot 2016-06-25 at 1 01 15 am](https://cloud.githubusercontent.com/assets/2678795/16352828/749b4be2-3a72-11e6-8346-dec3e09fd1c1.png)
where the label is not touching the shape at all. Because it's rare, doesn't actually stop the user from getting the information, and the fix would be really hairy, I propose to ignore it 🙈